x-pack/plugins/cases/public/components/case_view/components/case_view_activity.tsx

```diff
@@ -58,10 +58,19 @@ export const CaseViewActivity = ({
     [caseData.assignees]
   );
 
+  const [uidsToRetrieve, setUidsToRetrieve] = useState<string[]>([]);
+
   const { data: userProfiles, isLoading: isLoadingUserProfiles } = useBulkGetUserProfiles({
-    uids: assignees,
+    uids: uidsToRetrieve,
   });
 
+  useEffect(() => {
```

Comment:
There is no need to use a `useEffect` to derive the uids. You can do:

```
const uidsToRetrieve = uniq([...userActionsData?.profileUids ?? [], ...assignees]);

const { data: userProfiles, isLoading: isLoadingUserProfiles } = useBulkGetUserProfiles({
    uids: uidsToRetrieve,
  });
```

Reply:
ah ok, just to make sure I understand, let's say `useGetCaseUserActions` initially returns `undefined` because it's loading, we'll call `useBulkGetUserProfiles` with only the `assignees`. Once `useGetCaseUserActions` is finished loading, the component will rerender and it'll automatically call `useBulkGetUserProfiles` with the updated `uidsToRetrieve` if we don't include a `useEffect`?